PIVX-Labs/MyPIVXWallet
# Auto deshield transaction

## Abstract

This PR introduces auto deshielding.
Some places like exchanges don't support shield transactions. To circumvent this, MPW is set to automatically deshield to a newly generated address before sending. 
Closes #337

## Can this transaction be tracked to you?
The burner address is generated using a new change address, which MPW will not reuse. Tools like https://chainz.cryptoid.info/pivx/#!wallets will not work.
However these addresses can be easily tracked if the xpub is known. As of right now the xpub is advertised to explorers, while we only allow a set of trusted explorers, we can't be 100% sure they're not tracking user addresses, although very unlikely. If absolute privacy is needed, as always, it's best to self host an explorer.

We could technically make an address unlinked to the xpub or using a special hardened path, however this makes things more difficult to recover should the first deshield transaction go through and the second one fail.
If this were to happen right now, the deshielded funds would simply be available as regular balance.

## Other changes
Shield addresses will now show as red in transfermenu if shield is not enabled

## TODO
- [x] Figure out what happens when sending with auto shield enabled to a shield address

## Testing
- Send to a transparent address using auto shield. Two transactions should be created and the shield balance should be used
- Send to an exchange address/xpub
- Send to a shield address and test that it throws a user friendly error
